docs/commands.md

```diff
@@ -1,382 +1,625 @@
----
+!!! warning "Intermediate memory overhead"
+    The execution of scripts may generate intermediate tensors that are not allocated by the Redis allocator, but by whatever allocator is used in the backends (which may act on main memory or GPU memory, depending on the device), thus not being limited by `maxmemory` configuration settings of Redis.
 
 ## AI._SCRIPTLIST
+The **AI._SCRIPTLIST** command returns all the scripts in the database.
 
-NOTE: `_SCRIPTLIST` is EXPERIMENTAL and might be removed in future versions.
+!!! warning "Experimental API"
+    `AI._MODELLIST` is an EXPERIMENTAL command that may be removed in future versions.
 
-List all scripts. Returns a list of (key, tag) pairs.
+**Redis API**
 
-```sql
+```
 AI._SCRIPTLIST
 ```
 
-### _SCRIPTLIST Example
+_Arguments_
 
-```sql
-AI.SCRIPTSET script1 GPU TAG addtwo:v0.1 < addtwo.txt
-AI.SCRIPTSET script2 GPU TAG addtwo:v0.2 < addtwo.txt
+None.
 
-AI._SCRIPTLIST
+_Return_
+
+An array with an entry per script. Each entry is a an array with two entries:
+
+1. The script's key name as a String
+1. The script's tag as a String
+
+**Examples**
 
->  1) 1) "script1"
->  1) 2) "addtwo:v0.1"
->  2) 1) "script2"
->  2) 2) "addtwo:v0.2"
 ```
----
+redis> > AI._SCRIPTLIST
+1) 1) "myscript"
+   2) "myscript:v0.1"
+```
+
+## AI.DAGRUN
+The **`AI.DAGRUN`** command specifies a direct acyclic graph of operations to run within RedisAI. 
+It accepts one or more operations, split by the pipe-forward operator (`|>`). 
+
+
+By default, the DAG execution context is local, meaning that loading and persisting tensors should be done explicitly. The user should specify which key tensors to load from keyspace using the `LOAD` keyword, and which command outputs to persist to the keyspace using the `PERSIST` keyspace.
+
+When `PERSIST` is not present, object savings are done locally and kept only during the context of the DAG meaning that no output keys are open.
+
+As an example, if `command 1` sets a tensor, it can be referenced by any further command on the chaining. 
+
+
+**Redis API**
+
+```
+AI.DAGRUN [LOAD <number keys> <key 1> ... <key n>]
+          [PERSIST <number keys> <key 1> ... <key n>] |>
+          [command 1] |>  [command 2] |> [command n]
+```
+
+_Arguments_
+
+* **LOAD**: an optional argument, that denotes the beginning of the input tensors keys' list, followed by the number of keys, and one or more key names
+* **PERSIST**: an optional argument, that denotes the beginning of the output tensors keys' list, followed by the number of keys, and one or more key names
+* **|> command**: the chaining operator, that denotes the beginning of a RedisAI command, followed by one of the supported RedisAI commands ( for now `AI.TENSORGET`, `AI.TENSORGET`, and `AI.MODELRUN` ). Command spliting is done by the presence of another `|>`.
```

Comment:
Repeating `AI.TENSORGET`

Reply:
solved in latest commit. thanks @hhsecond